src/lightning/pytorch/strategies/xla_fsdp.py

```diff
@@ -456,27 +468,37 @@ def load_checkpoint(self, checkpoint_path: _PATH) -> Dict[str, Any]:
 
         from torch_xla.distributed.fsdp import XlaFullyShardedDataParallel as XLAFSDP
 
-        if self.model is None or not isinstance(self.model, XLAFSDP):
-            raise ValueError(f"Could not find a XLAFSDP model in the provided checkpoint state: {type(self.model)}")
-
         if self._state_dict_type == "sharded":
             file = path / f"checkpoint_rank-{self.global_rank:08d}-of-{self.world_size:08d}.pth"
             if not file.is_file():
                 raise ValueError(
                     f"The path {str(file)!r} does not point to valid sharded checkpoints. Make sure the path points to"
                     " a directory with XLAFSDP checkpoint shards."
                 )
+            if self.model is None or not isinstance(self.model, XLAFSDP):
+                raise ValueError(f"Could not find a XLAFSDP model in the provided checkpoint state: {type(self.model)}")
             sharded_ckpt = torch.load(file)
             self.model.load_state_dict(sharded_ckpt.pop("model"))
-            sharded_ckpt.pop("shard_metadata", None)
```

Comment:
Don't we need to pop this? Sharded checkpoints would have this field still, no?

Reply:
It doesn't matter since we are no longer returning `sharded_ckpt` the code change is

```
return sharded_ckpt
```

to 

```
# Load metadata (anything not a module or optimizer)
metadata = torch.load(path / _METADATA_FILENAME)
return metadata
 ```
 
 Ref: https://github.com/Lightning-AI/lightning/pull/18746/files#diff-801cfb5c353fa5764dfbc0d1e1a0cdd4a1cacef116fc2b6a07cb7c2dc3b5d71cR483-R485